File: tests/csharp/collections_advanced/collection_initializer_list.cs
// vybe-test: csharp/collections_advanced/collection_initializer_list
// origin: languages/csharp/tests/csharp/test_collections_advanced.rs

using static __Harness;

var names = new List<string> { "Alice", "Bob", "Charlie" }
;
__P((names.Count).ToString());
__P((names[1]).ToString());
__Check("3\nBob");

public static class __Harness {
    public static string __buf = "";
    public static void __P(string s) { __buf = __buf + s + "\n"; }
    public static void __Pr(string s) { __buf = __buf + s; }
    public static void __Check(string want) {
        if (__buf != want && __buf != want + "\n") {
            Console.WriteLine("FAIL: want [" + want + "] got [" + __buf + "]");
            throw new Exception("assertion failed");
        }
    }
}

File: tests/vb/vb_collection_initializers_list/collection_initializer_list.vb
' vybe-test: vb/vb_collection_initializers_list/collection_initializer_list
' origin: languages/vb/tests/vb/test_vb_collection_initializers_list.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Imports System.Collections.Generic

Module M
    Sub Main()
        ' Collection initializer syntax
        Dim list As New List(Of String) From {"apple", "banana", "cherry"}
        
        __Check(CStr(list.Count), "3")
        __Check(CStr(list(1)), "banana")
    End Sub
End Module
